lib/gemstash/http_client.rb

```diff
@@ -14,7 +14,13 @@ def initialize(message, code)
   end
 
   #:nodoc:
+  class ConnectionError < StandardError
```

Comment:
What do you think of:
```
class ConnectionError < Gemstash::WebError
  def initialize(message)
    super(message, 502) # Bad Gateway
  end
end
```

Then the `rescue Gemstash::WebError => e` in `upstream_source.rb`  which halts with the WebError code would capture both conditions. Also, I think a `Gemstash::ConnectioError` could arguably be considered a form of a WebError.